### ISSUE TITLE
Common key remappings for both normal and select mode

### ISSUE THREAD

**Issue by larsblumberg** [helix-editor/helix#5106]
My `config.toml` contains many key remappings that are identical for both `normal` and `select` mode, example:

```
[keys.normal]
"^" = "goto_line_start"
"$" = "goto_line_end"
"_" = "goto_first_nonwhitespace"
G = "goto_file_end"
# Always copy to system clipboard, too
d = ["yank_joined_to_clipboard", "yank", "delete_selection"]
y = ["yank_joined_to_clipboard", "yank"]
"space" = { "space" = "file_picker_in_current_directory" }
"C-h" = "jump_view_left"
"C-j" = "jump_view_down"
"C-k" = "jump_view_up"
"C-l" = "jump_view_right"
"C-q" = "wclose"
"ret" = "open_below"
"A-ret" = "open_above"

[keys.select]
"^" = "goto_line_start"
"$" = "goto_line_end"
"_" = "goto_first_nonwhitespace"
G = "goto_file_end"
# Always copy to system clipboard, too
d = ["yank_joined_to_clipboard", "yank", "delete_selection"]
y = ["yank_joined_to_clipboard", "yank"]
"space" = { "space" = "file_picker_in_current_directory" }
"C-h" = "jump_view_left"
"C-j" = "jump_view_down"
"C-k" = "jump_view_up"
"C-l" = "jump_view_right"
"C-q" = "wclose"
```

Using above example, a hypothetical, not yet existing `[mode.normal_select]` section could unify common key remappings:

```
# shared key remappings for both normal and select
[keys.normal_select]
"^" = "goto_line_start"
"$" = "goto_line_end"
"_" = "goto_first_nonwhitespace"
G = "goto_file_end"
# Always copy to system clipboard, too
d = ["yank_joined_to_clipboard", "yank", "delete_selection"]
y = ["yank_joined_to_clipboard", "yank"]
"space" = { "space" = "file_picker_in_current_directory" }
"C-h" = "jump_view_left"
"C-j" = "jump_view_down"
"C-k" = "jump_view_up"
"C-l" = "jump_view_right"
"C-q" = "wclose"

# normal mode only key remappings
[mode.normal]
"ret" = "open_below"
"A-ret" = "open_above"

# select mode only key remappings
[mode.select]
# none currently
```

Would others find it useful, too, if common remappings could be defined only once?

**Comment by the-mikedavis** [comment]
Duplicate of https://github.com/helix-editor/helix/issues/2125
